phd-tests/framework/src/guest_os/mod.rs

```diff
@@ -11,30 +11,32 @@ use serde::{Deserialize, Serialize};
 
 mod alpine;
 mod debian11_nocloud;
+mod shell_commands;
 mod ubuntu22_04;
 mod windows;
+mod windows_server_2016;
 mod windows_server_2019;
 mod windows_server_2022;
 
 /// An entry in a sequence of interactions with the guest's command prompt.
 #[derive(Debug)]
-pub(super) enum CommandSequenceEntry {
+pub(super) enum CommandSequenceEntry<'a> {
     /// Wait for the supplied string to appear on the guest serial console.
-    WaitFor(&'static str),
+    WaitFor(Cow<'a, str>),
```

Comment:
This feels like it might be the wrong tool for the job, but maybe not?

Previously this was only used for the string literals that appear in guest OSes' boot sequences, but now a command *might* be an owned string, e.g. if `shell_command_sequence` had to amend it. It seemed silly to make all the literals into owned `String`s just to accommodate this case, but I'm not sure if `Cow` is the best (or even an appropriate) way for this code to have its `&'static str` cake and eat `String`s too.

Reply:
`Cow` is absolutely an appropriate way to have something that's "either a `&'static str` or a `String`" --- honestly, I think I see it used for that purpose far more than for use-cases that actually use copy-on-write...